src/main/java/io/codelex/flightplanner/FlightService.java

```diff
@@ -0,0 +1,91 @@
+package io.codelex.flightplanner;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+public class FlightService {
+    private long id = 0;
+    private final FlightRepository flightRepository;
+
+    public FlightService(FlightRepository flightRepository) {
+        this.flightRepository = flightRepository;
+    }
+
+    public void clearFlights() {
+        flightRepository.clearFlights();
+    }
+
+    public PageResult searchFlightByFromToAndDepartureDate(SearchFlightsRequest flight) {
+        return flightRepository.searchFlightByFromToAndDepartureDate(flight);
+    }
+
+    public List<Airport> searchAirport(String search) {
+        List<Airport> searchResults = new ArrayList<>();
+
+        Set<Airport> allAirports = flightRepository.getFlights().stream()
+                .map(Flight::getFrom)
+                .collect(Collectors.toSet());
+
+        for (Airport airport : allAirports) {
+            if (airport.textForSearch().contains(search.trim().toLowerCase())) {
+                searchResults.add(airport);
+            }
+        }
+        return searchResults;
+    }
+
+    public synchronized void deleteFlightById(long id) {
```

Comment:
Šo arī pārnest uz repository (lai vēlāk vieglāk), noņem šeit synchronized un liec to klāt pie repository metodes.